src/qforte/abc/algorithm.py

```diff
@@ -53,22 +53,39 @@ class Algorithm(ABC):
     def __init__(self,
                  system,
                  reference=None,
-                 trial_state_type='reference',
+                 state_prep_type='occupation_list',
                  trotter_order=1,
                  trotter_number=1,
                  fast=True,
                  verbose=False,
                  print_summary_file=False):
 
         self._sys = system
-        if(reference==None):
+        self._state_prep_type = state_prep_type
+
+        if self._state_prep_type == 'occupation_list':
+            if(reference==None):
+                self._ref = system.get_hf_reference()
+            else:
+                if not (isinstance(reference, list)):
```

Comment:
Do we ever validate that the reference is of the correct length?

Reply:
These lines accomplish this I think.

```
self._nqb = len(self._ref)
self._qb_ham = system.get_hamiltonian()
if self._qb_ham.num_qubits() != self._nqb:
     raise ValueError(f"The reference has {self._nqb} qubits, but the Hamiltonian has {self._qb_ham.num_qubits()}. This is inconsistent.")
```